dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/plugins/RuntimePerspectivePluginManager.java

```diff
@@ -0,0 +1,131 @@
+import org.dashbuilder.navigation.layout.LayoutTemplateContext;
+import org.dashbuilder.navigation.layout.LayoutTemplateInfo;
+import org.dashbuilder.navigation.workbench.NavWorkbenchCtx;
+import org.dashbuilder.shared.event.RuntimeModelEvent;
+import org.jboss.errai.common.client.ui.ElementWrapperWidget;
+import org.uberfire.ext.layout.editor.api.editor.LayoutInstance;
+import org.uberfire.ext.layout.editor.api.editor.LayoutTemplate;
+import org.uberfire.ext.layout.editor.client.generator.LayoutGenerator;
+import org.uberfire.ext.plugin.model.Plugin;
+import org.uberfire.ext.plugin.model.PluginType;
+import org.uberfire.mvp.ParameterizedCommand;
+
+@Alternative
+@ApplicationScoped
+public class RuntimePerspectivePluginManager implements PerspectivePluginManager {
+    
+    @Inject
+    LayoutGenerator layoutGenerator;
+
+    List<LayoutTemplate> templates = new ArrayList<>();
+
+    @Override
+    public void loadPlugins() {
+
+    }
+
+    @Override
+    public void getPerspectivePlugins(ParameterizedCommand<Collection<Plugin>> callback) {
+        List<Plugin> plugins = templates.stream()
+                                        .map(lt -> new Plugin(lt.getName(),
+                                                              PluginType.PERSPECTIVE,
+                                                              null))
+                                        .collect(Collectors.toList());
+        callback.execute(plugins);
+    }
+
+    @Override
+    public boolean isRuntimePerspective(Plugin plugin) {
+        return searchLayoutTemplate(plugin.getName());
+    }
+
+    @Override
+    public boolean isRuntimePerspective(NavItem navItem) {
+        NavWorkbenchCtx navCtx = NavWorkbenchCtx.get(navItem);
+        String resourceId = navCtx.getResourceId();
+        return searchLayoutTemplate(resourceId);
+    }
+
+    @Override
+    public boolean isRuntimePerspective(String perspectiveId) {
+        return searchLayoutTemplate(perspectiveId);
+    }
+
+    @Override
+    public String getRuntimePerspectiveId(NavItem navItem) {
+        NavWorkbenchCtx navCtx = NavWorkbenchCtx.get(navItem);
+        return navCtx.getResourceId();
+    }
+
+    @Override
+    public boolean existsPerspectivePlugin(String perspectiveName) {
+        return searchLayoutTemplate(perspectiveName);
+    }
+
+    @Override
+    public void getLayoutTemplateInfo(String perspectiveName, ParameterizedCommand<LayoutTemplateInfo> callback) {
+        // not used in runtime
+    }
+
+    @Override
+    public void getLayoutTemplateInfo(LayoutTemplate layoutTemplate, ParameterizedCommand<LayoutTemplateInfo> callback) {
+        // not used in runtime
+    }
+
+    @Override
+    public void buildPerspectiveWidget(String perspectiveName, LayoutTemplateContext layoutCtx, ParameterizedCommand<IsWidget> afterBuild, ParameterizedCommand<LayoutRecursionIssue> onInfiniteRecursion) {
+        // LAYOUT RECURSION SHOULD BE HANDLED HERE
```

Comment:
Could you please clarify?

Reply:
This layout recursion should not be a problem after authoring, but I left a comment to remember where to fix if we face it.